### DISCUSSION TITLE
Mask/Remove Voxels in Air (small intersection with surface)

### DISCUSSION THREAD

**Discussion by abbas-tari** [pyvista/pyvista#2617] [Q&A]
Hi all - My question is about the `voxelize` function. I do the voxelization for a simple shape and in the end, there are some voxels which only small portion of them touches the surface and the most volume is the air as it is shown below. **Is there any way we could remove/mask (or detect) these voxels as they only have very small intersection with the surface?**

<img src="https://user-images.githubusercontent.com/4333912/137623030-5b2be65c-4d1c-49fa-8957-6bc8097933a6.png" width=35% height=35%>

**Comment by RichardScottOZ** [comment]
If your surface is a plane, you can clip to that - so you can end up with a dataset either way.

**Comment by abbas-tari** [comment]
Thanks. I wasn't aware of clipping.  I solved it with another way in which if center points of the voxels aren't enclosed within the surface, they will be discarded.  This was enough for me, however your approach is more elegant.

**Comment by RichardScottOZ** [comment]
Ok, good. You can close this one?
